### PR TITLE
Added remain Maven API, update test

### PR DESCRIPTION
**Pull request by shblue21**
# Azure DevOps java sdk pull request template guide

Thank you for your interest in contributing to this project.

## Description

I implemented the remaining maven api. (Excluding download). Test was implemented for all cases, and the entire test function was also written.

## PR Checklist

- [x] Added help
- [x] Added unit test/s
- [x] Updated Change log

### CODE REVIEW

**Review by hkarthik7 — CHANGES_REQUESTED** [review]
Here are the test failure messages:

[ERROR]   MavenApiTest.shouldGetPackageVersion:63 expected:<org.[HelloWorld:HelloW]orld> but was:<org.[helloworld:hellow]orld>
[ERROR]   MavenApiTest.shouldGetPackageVersionWithQueryParameters:72 expected:<org.[HelloWorld:HelloW]orld> but was:<org.[helloworld:hellow]orld>

**Comment by hkarthik7** [comment]
@shblue21 no luck mate. Worth updating the tests once and then give it a try.

**Comment by shblue21** [comment]
@hkarthik7 First of all, sorry for generating a lot of release pipeline logs.
I modified the test code and created an example document.
Please let me know if you need to edit the test. (Delete sysout, etc.)

**Comment by shblue21** [comment]
And I can't implement DownloadPackage right away. Because

**Comment by hkarthik7** [comment]
**@shblue21** The download functionality has to be tested intensely before applying. We should think a way to introduce this without affecting the core functionality of SDK. After all we've built this for our use and can come up with a way that works for everyone. Since few projects depends on the SDK for Azure DevOps integration I would recommend to add the `download package` functionality later.

You can write an exact API wrapper for `downloadPackage` and create a helper method that calls `downloadPackage` to actually download the contents in a `.zip` or any format that makes sense.

[Description from documentation](https://docs.microsoft.com/en-us/rest/api/azure/devops/artifactspackagetypes/maven/download-package?view=azure-devops-rest-7.1)  - 
*Fulfills Maven package file download requests by either returning the URL of the requested package file or, in the case of Azure DevOps Server (OnPrem), returning the content as a stream.*

I did a quick test from Postman and the response was a stream content.

So, you can create a method `downlodPackage` that returns file stream. Then create a helper method something like `downloadPackageContent` and ask the user to choose the file format. It can be a text, json, .md, .zip etc, because it expected that user should be aware of file format they're requesting for. An example logic shown [here](https://stackoverflow.com/questions/27641774/how-to-download-zip-file-from-url-using-java)

**Review by hkarthik7 — COMMENTED** [review]
LGTM

File: azd/src/main/java/org/azd/maven/MavenApi.java [hunk 1]
```diff
@@ -210,18 +214,34 @@ public void deletePackageVersionFromRecycleBin(String feedId, String groupId, St
      * @param groupId    Group ID of the package. Example: "com.example".
      * @param artifactId Artifact ID of the package. Example: "app".
      * @param version    Version of the package. Example: "1.0.0".
-     * @param promote    State of the package. Example: "prelease".
+     * @param promote    State of the package. Example: "prelease". {@link MavenPackagePromote}
      * @throws AzDException Default Api Exception handler.
      */
     @Override
     public void updatePackageVersion(String feedId, String groupId, String artifactId, String version, MavenPackagePromote promote)
             throws AzDException {
```
[the remark this answers is not in the document]
Comment (shblue21) [reply azd/src/main/java/org/azd/maven/MavenApi.java:222]: As you know, you can create a Custom View type in Azure Devops.

**Review by hkarthik7 — APPROVED** [review]
LGTM

**Comment by hkarthik7** [comment]
Happy to do a squash merge.

**Comment by shblue21** [comment]
Thanks for merging!

**Comment by hkarthik7** [comment]
@shblue21, I've modified the `downloadPackage` method in **MavenApi**. Please take a look and test it, if the implementation is not what you're looking for or you didn't like please feel free to modify it.

**Comment by shblue21** [comment]
@hkarthik7   mate. I'll check right away. thank you very much.

**Comment by shblue21** [comment]
@hkarthik7  It seems to be working fine.

**Comment by shblue21** [comment]
@hkarthik7 I want to implement uploadPackage. 
The request url has the following format:  
PUT pkgs.dev.azure.com/{account}/{org}/**_packaging**/{feed}/maven/v1/{artifactid}/{groupid}/{version}/{filename}  

But the corresponding url for API_RELATIVE_PATH is _packaging .
Is there any way to implement this by changing API_RELATIVE_PATH ?


```java
public final class RestClientProvider extends BaseRestClient {
    protected static final JsonMapper MAPPER = new JsonMapper();
    private static final String API_RELATIVE_PATH = "_apis";
```

**Comment by hkarthik7** [comment]
@shblue21, please implement what makes sense to you. I don’t recommend modifying the existing relative path property as it will change the entire functionality of the library. You can do something like creating a string build to construct the url and call RestClient.send() that demands a request url.

**Pull request merged**